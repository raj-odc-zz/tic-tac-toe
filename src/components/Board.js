import React from 'react';
import Field from './Field';

const fields = [0, 1, 2, 3, 4, 5, 6, 7, 8];

const Board = () => (
  <div className="Board">
    {fields.map(id => <Field id={id} key={id} />)}
  </div>
);

export default Board;
