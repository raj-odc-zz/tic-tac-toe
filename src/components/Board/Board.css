.Board {
  position: relative;

  width: 100%;
  height: 100%;
}

.nextPlayer {
  font-weight: 600;
  font-size: 14px;
  margin-left: 5px;
}

.winner__section {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  font-weight: 600;
}

.winner__section--text {

  display: flex;
  flex-direction: row;
  justify-content: center;
}