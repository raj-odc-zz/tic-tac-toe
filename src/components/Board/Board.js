import React, { Component } from 'react';
import { connect } from 'react-redux';

import Square from '../Square/Square';

import './Board.css';

import { createBoard, updateSquareAndPlayer } from '../../store/actions';

const squares = [0, 1, 2, 3, 4, 5, 6, 7, 8];

class Board extends Component {
  componentDidMount() {
    const { initiateBoard } = this.props;
    initiateBoard();
  }

  handleClick = (id) => {
    const { onSquareClick, winner } = this.props;
    if (winner) return;
    onSquareClick(id);
  }

  render() {
    const {
      activePlayer, board, initiateBoard, players, winner,
    } = this.props;
    return (
      <React.Fragment>
        {
          winner ? (
            <div className="winner__section">
              <div className="winner__section--text">
                { winner === 'Draw' ? 'Match Drawn' : `Player ${winner} won` }
              </div>
              <button type="button" className="winner__section--restart" onClick={initiateBoard}>Play Again</button>
            </div>
          )
            : (
              <div className="player__section">
                Next Player:
                {players[activePlayer]}
              </div>
            )
        }
        <div className="Board">
          {
            squares.map(id => (
              <Square
                onClick={() => this.handleClick(id)}
                squareValue={board[id]}
                squareIndex={id}
                key={id}
              />
            ))
          }
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => ({
  activePlayer: state.activePlayer,
  board: state.board,
  winner: state.winner,
  players: state.players,
});

const mapDispatchToProps = dispatch => ({
  onSquareClick: squareIndex => dispatch(updateSquareAndPlayer(squareIndex)),
  initiateBoard: () => dispatch(createBoard()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Board);
