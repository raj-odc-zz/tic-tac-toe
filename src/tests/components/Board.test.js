import * as React from 'react';
import { mount } from 'enzyme';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

import Board from '../../components/Board/Board.js';

describe('<Board /> with players', () => {

  let wrapper;
  const middlewares = [ thunk ];
  
  let initialState = {
    board: [null, null, null, null, null, null, null, null, null],
    activePlayer: 1,
    winner: null,
    players: {1: 'John', 2: 'Ashok'},
  };
  const mockStore = configureStore(middlewares);
  const store = mockStore(initialState);

  it('should load the next player as John',() => {
    wrapper = mount(<Board store={store}/>);
    expect(wrapper.find('div.player__section').text()).toEqual('Next Player:John');
  });

  it('should load the next player as Ashok',() => {
    initialState.activePlayer = 2;
    const wrapper = mount(<Board store={store}/>);
    expect(wrapper.find('div.player__section').text()).toEqual('Next Player:Ashok');
  });

  it('should load the winner section when state has winner',() => {
    initialState.winner = "Ashok";
    const wrapper = mount(<Board store={store}/>);
    expect(wrapper.find('div.winner__section--text').text()).toEqual('Player Ashok won');
  });
});