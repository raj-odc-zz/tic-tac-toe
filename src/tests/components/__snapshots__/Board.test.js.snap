// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> with players renders correctly 1`] = `
Array [
  <div
    className="player__section"
  >
    Next Player:
    John
  </div>,
  <div
    className="Board"
  >
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="Field"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
  </div>,
]
`;
